src/mqt/predictor/rl/tracer.py

```
@@ -0,0 +1,292 @@
+    def record_step(
+        self,
+        step_index: int,
+        action: str,
+        reward: float,
+        current_qc: QuantumCircuit,
+        expected_fidelity: float,
+        fidelity_kind: str,
+        fom_value: float,
+        fom_kind: str,
+        features: dict[str, int | NDArray[np.float32]],
+        synthesized: bool,
```

Ruff (0.15.9)

[warning] Boolean-typed positional argument in function definition

(FBT001)